source/features/copy-file.tsx

```diff
@@ -3,28 +3,49 @@ import select from 'select-dom';
 import copyToClipboard from 'copy-text-to-clipboard';
 import features from '../libs/features';
 
-function handleClick({currentTarget: button}: React.MouseEvent<HTMLButtonElement>): void {
-	const file = button.closest('.Box');
+function handleClickFactory(markdown?: Element): ({currentTarget: button}: React.MouseEvent<HTMLButtonElement>) => void {
+	return function ({currentTarget: button}: React.MouseEvent<HTMLButtonElement>): void {
+		if (features.isMarkDown()) {
+			const content = select.all('.blame-hunk', markdown)
+				.map(line => select.all('.blob-code', line))
+				.map(line => {
+					return line.map(x => x.innerText).map(line => line === '\n' ? '' : line).join('\n');
+				})
+				.map(line => line === '\n' ? '' : line)
+				.join('\n');
+			copyToClipboard(content);
+		} else {
+			console.log('not markdown');
+			const file = button.closest('.Box');
 
-	const content = select.all('.blob-code-inner', file!)
-		.map(blob => blob.innerText) // Must be `.innerText`
-		.map(line => line === '\n' ? '' : line)
-		.join('\n');
+			const content = select.all('.blob-code-inner', file!)
+				.map(blob => blob.innerText) // Must be `.innerText`
+				.map(line => line === '\n' ? '' : line)
+				.join('\n');
 
-	copyToClipboard(content);
+			copyToClipboard(content);
+		}
+	};
 }
 
-function init(): void {
-	// This selector skips binaries + markdowns with code
-	for (const code of select.all('.blob-wrapper > .highlight:not(.rgh-copy-file)')) {
-		code.classList.add('rgh-copy-file');
+async function init(): Promise<void> {
+	// This won't show button on markdown and binary files until the code is loaded
+	if (features.isMarkDown() || select.all('.blob-wrapper > .highlight').length !== 0) {
+		document.addEventListener('rgh:view-markdown-source', e => {
```

Comment:
You can use
```suggestion
		delegate('.rgh-md-source', 'rgh:view-markdown-source', event => {
			renderButton(event.delegateTarget);
```

Reply:
this results in type error since `rgh:...` isn't an event type. I'm not sure that it can be overridden.

Reply:
You can find the events in `global.d.ts`, next to `details:toggle`